pipeline_steps/common.groovy

```diff
@@ -1438,3 +1438,104 @@ Boolean isSkippable(String skip_pattern, String credentials) {
   }
   return skipIt
 }
+
+void testReleasesPullRequest(String baseBranch, String prBranch){
+  venv = "${WORKSPACE}/.componentvenv"
+  sh """#!/bin/bash -xe
+      virtualenv --python python3 ${venv}
+      set +x; . ${venv}/bin/activate; set -x
+      pip install -c '${env.WORKSPACE}/rpc-gating/constraints_rpc_component.txt' rpc_component
+  """
+  types = ["release", "registration"]
+  for (i=0; i < types.size(); i++){
+    type = types[i]
+    try{
+      component_text = sh(
+        script: """#!/bin/bash -xe
+          set +x; . ${venv}/bin/activate; set -x
+          component --releases-dir . compare --from ${baseBranch} --to ${prBranch} --verify ${type}
+        """,
+        returnStdout: true
+      )
+      break
+    }catch (e){
+      println "Pull request is not a ${type}."
+      println e
+      component_text = null
+    }
+  }
+  if (component_text == null) {
+    throw new Exception("The pull request does not correspond to any of ${types}.")
+  }
+
+  println "Pull request is a ${type}."
+  println "=== component CLI standard out ==="
+  println component_text
+  if (type == "release"){
+    testRelease(component_text)
+  }else{
+    throw new Exception("Unknown pull request type ${type}")
```

Comment:
`types` array seems to have two valid values, but we only test for "release".  Perhaps this exception should say "Pull request type ${type} currently not supported" as the previous loop and null check identifies types that are not defined in the array.

Reply:
Reworded.

Good spot, I missed this when I rewrote my first version.